api/src/main/java/com/github/bsideup/liiklus/records/RecordsStorage.java

```diff
@@ -28,6 +37,79 @@
         );
     }
 
+    default Wire<ByteBuffer, String, String> toWire(Envelope envelope) throws IllegalArgumentException {
+        Object rawValue = envelope.getRawValue();
+
+        if (!(rawValue instanceof CloudEvent)) {
+            // TODO Add Envelope#event and make CloudEvent a fist-class citizen
+            throw new IllegalArgumentException("Must be a CloudEvent!");
```

Comment:
Trying this PR in the context of projectriff, it seems I'm always hitting this line. I'm sure I'm doing something wrong, but I don't know what yet.

Reply:
make sure you publish with the new protocol (and set the `liiklusEvent` field)